hybridse/include/base/fe_status.h

```diff
@@ -39,82 +40,81 @@ static inline std::initializer_list<int> __output_literal_args(STREAM& stream,
-        }                                                                                            \
-        break;                                                                                       \
+#define CHECK_TRUE(call, errcode, ...)                                  \
+    while (true) {                                                      \
+        if (!(call)) {                                                  \
+            std::stringstream _msg;                                     \
+            hybridse::base::__output_literal_args(_msg, ##__VA_ARGS__); \
+            hybridse::base::Status _status(errcode, _msg.str());        \
+            _status.AddTrace(__FILE__, __LINE__, _msg.str());           \
+            return _status;                                             \
+        }                                                               \
+        break;                                                          \
     }
-#define FAIL_STATUS(errcode, ...)                                                                \
-    while (true) {                                                                               \
-        std::stringstream _msg;                                                                  \
-        hybridse::base::__output_literal_args(_msg, ##__VA_ARGS__);                              \
-        std::stringstream _trace;                                                                \
-        hybridse::base::__output_literal_args(_trace, "    (At ", __FILE__, ":", __LINE__, ")"); \
-                                                                                                 \
-        if (!_msg.str().empty()) {                                                               \
-            _trace << "\n"                                                                       \
-                   << "    (Caused by) " << _msg.str();                                          \
-        }                                                                                        \
-        hybridse::base::Status _status(errcode, _msg.str(), _trace.str());                       \
-        return _status;                                                                          \
-        break;                                                                                   \
+#define FAIL_STATUS(errcode, ...)                                   \
+    while (true) {                                                  \
+        std::stringstream _msg;                                     \
+        hybridse::base::__output_literal_args(_msg, ##__VA_ARGS__); \
+                                                                    \
+        hybridse::base::Status _status(errcode, _msg.str());        \
+        _status.AddTrace(__FILE__, __LINE__, _msg.str());           \
+        return _status;                                             \
```

Comment:
is this `_status` returned a copy of previous one ? 
if so can we reuse single `Status` instance in fail chain ?

Reply:
Yep. CHECK_STATUS will reuse single `Status` instance in fail chain.
`CHECK_TRUE` and `FAIL_STATUS` will create a new instance